### DISCUSSION TITLE
NARX for Classification using MetaMSS

### DISCUSSION THREAD

**Discussion by andreascrocca** [wilsonrljr/sysidentpy#150] [Q&A] [answered]
I can't find examples on the repository regarding classification problems which have been addressed using NARX models developed with metaMSS. Is it possible to look at one? Also logistic-NARX models would work.

**Comment by wilsonrljr** [comment, marked as answer]
I haven't implemented MetaMSS in SysIdentPy yet. Actually, I haven’t implemented any algorithms focused on classification so far. It’s in my future plans, but I imagine it will come around version 1.0 of the package. Since I develop everything on my own, it might take a while.  

However, I believe you can use the `general_estimators` method to create NARX models for classification. If you’re interested in this approach, I’d be happy to help you test it.

**Reply by andreascrocca** [reply]
I am trying to predict a binary variable based on 17 continuous variables; the problem is similar to the classification problem you face in the article "Meta-Model Structure Selection: Building Polynomial NARX Model for Regression and Classification".
So far I use NARMAX models to predict the output as I have a regression problem and then I classify the output =1 if it is greater than 0.5 and =0 viceversa.
I have tried with Narmax models and FROLS as MSS method but I can't obtain decent results. With the general_estimators I get the following error 'ValueError: X has 2701 features, but GradientBoostingRegressor is expecting 253 features as input.'

What would you suggest?
